views/components/ChartList.jsx

```diff
@@ -45,54 +51,117 @@ const ChartList = () => {
-            </TableCell>
-            <TableCell align='center'>{description?.toUpperCase()}</TableCell>
-            <TableCell align='center'>
-              <Button
-                type='button'
-                variant='text'
-                onClick={() => onDuplicateChart(_id)}
-              >
-                <PlaylistAdd />
-              </Button>
-            </TableCell>
-            <TableCell align='center'>
-              <Link href={routes.editChart(_id)} color='textPrimary'>
-                <Edit />
-              </Link>
-            </TableCell>
-            <TableCell align='center'>
-              <Button
-                type='button'
-                variant='text'
-                onClick={() => removeChart(_id)}
-              >
-                <Delete />
-              </Button>
-            </TableCell>
+    <>
+      <Table>
+        <TableHead>
+          <TableRow>
+            <TableCell align='center'>Title</TableCell>
+            <TableCell align='center'>Description</TableCell>
+            <TableCell align='center'>Duplicate</TableCell>
+            <TableCell align='center'>Edit</TableCell>
+            <TableCell align='center'>Share</TableCell>
+            <TableCell align='center'>Delete</TableCell>
           </TableRow>
-        ))}
-      </TableBody>
-    </Table>
+        </TableHead>
+        <TableBody>
+          {chartList.map((chart) => (
+            <TableRow key={chart._id}>
+              <TableCell align='center'>
+                <Link color='textPrimary' href={routes.chart(chart._id)}>
+                  {chart.title?.toUpperCase()}
+                </Link>
+              </TableCell>
+              <TableCell align='center'>
+                {chart.description?.toUpperCase()}
+              </TableCell>
+              <TableCell align='center'>
+                <Button
+                  type='button'
+                  variant='text'
+                  onClick={() => onDuplicateChart(chart._id)}
+                >
+                  <PlaylistAdd />
+                </Button>
+              </TableCell>
+              <TableCell align='center'>
+                <Link href={routes.editChart(chart._id)} color='textPrimary'>
+                  <Edit />
+                </Link>
+              </TableCell>
+              <TableCell align='center'>
+                <Button
+                  type='button'
+                  variant='text'
+                  onClick={() => handleShareChart(chart)}
+                >
+                  <Share />
+                </Button>
+              </TableCell>
+              <TableCell align='center'>
+                <Button
+                  type='button'
+                  variant='text'
+                  onClick={() => removeChart(chart._id)}
+                >
+                  <Delete />
+                </Button>
+              </TableCell>
+            </TableRow>
+          ))}
+        </TableBody>
+      </Table>
+      <ShareChart
```

Comment:
does this need to be in the `ChartList` component? It'd be nice if we could render it directly from the view so we don't need to pass the props down through multiple components.

Reply:
https://github.com/AMP-SCZ/dpdash/pull/131/commits/e250b027af6997da38ebfdf1d05fb1a75107db72